lib/ruby_lsp/requests/support/source_uri.rb

```diff
@@ -0,0 +1,82 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "uri/file"
+
+module URI
+  # Must be kept in sync with the one in Tapioca
```

Comment:
Why does it need to be kept in sync? Why can't we omit the methods that we don't use in `ruby-lsp`?